test/no-function-overriding.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @fileoverview Test for no-function-overriding rule
+ * @author Nicolas Feignon <nfeignon@gmail.com>
+ */
+
+'use strict';
+
+var Solium = require ('solium');
+
+var userConfig = {
+    rules: {
+        "security/no-function-overriding": "error"
+    }
+};
+
+describe ('[RULE] no-function-overriding: Acceptances', function () {
+
+    it ('should accept contracts that don\'t override their inherited functions', function (done) {
+        var code = [
+            `
+                contract A {
+                    function foo();
+                    function ();
+                }
+                contract B is A {
```

Comment:
please account for multiple inheritance also (eg- Contract C inherits from both A & B, then overrides something from A. Does it get flagged?) if its not already accounted for and add tests.

Plus is it possible for us to inherit a contract that doesn't exist in the same source file? If yes, please account for it too

Reply:
> please account for multiple inheritance also (eg- Contract C inherits from both A & B, then overrides something from A. Does it get flagged?) if its not already accounted for and add tests.

Forgot this case... Added some tests :)

> Plus is it possible for us to inherit a contract that doesn't exist in the same source file? If yes, please account for it too

Yes, it's possible to inherit a contract that doesn't exist in the same source file. Can I test that with Solium though? And should we do that in this test?

Reply:
Ah, so currently solium has a limitation that at any given time, a rule only has the context of a single file (so we're not aware of other files) - https://github.com/duaraghav8/Solium/issues/83

Just make sure that in such a case, the rule simply doesn't raise any issues since currently it can't read another file so can't determine whether there is any overriding or not. (I think this is naturally happening, but just make sure)

I'll fix this problem in future. Right now its good enough if we're only able to detect overriding when both contracts are in same file.

So please ensure the above, then we're good to merge!

Reply:
There was an issue in my code that would have raised an exception if we tried to inherit from a contract that is not defined in our code.
I fixed it here and in the no-inheritance PR. I also rebased this PR on master.